File: scratch/inspect_updated_bookings.go
package main

import (
	"database/sql"
	"fmt"
	"log"

	_ "github.com/glebarez/go-sqlite"
)

func main() {
	dbPath := "C:/Users/hofma/GolandProjects/HuhnLite-Wails/build/bin/HuhnLite.db"
	db, err := sql.Open("sqlite", dbPath)
	if err != nil {
		log.Fatalf("Error opening DB: %v", err)
	}
	defer db.Close()

	rows, err := db.Query(`
		SELECT ID, ID_HERDEN, BUCHUNGSDATUM, SILONR, FUTTERVERBRAUCHTIER, FUTTERKTAG, TIERBESTAND 
		FROM BUCHUNG 
		WHERE SILONR = 1 AND (FUTTERVERBRAUCHTIER > 0 OR FUTTERKTAG > 0)
		ORDER BY BUCHUNGSDATUM DESC LIMIT 10`)
	if err != nil {
		log.Fatal(err)
	}
	defer rows.Close()

	fmt.Println("=== BUCHUNG rows for Silo 1 in build/bin/HuhnLite.db ===")
	for rows.Next() {
		var id, idHerden, silonr, futterverbrauch, futterktag, tierbestand int64
		var datum string
		if err := rows.Scan(&id, &idHerden, &datum, &silonr, &futterverbrauch, &futterktag, &tierbestand); err == nil {
			fmt.Printf("ID: %d | Herd: %d | Date: %s | Silo: %d | Verbrauch: %d | Ktag: %d | Bestand: %d\n",
				id, idHerden, datum, silonr, futterverbrauch, futterktag, tierbestand)
		} else {
			fmt.Printf("Scan error: %v\n", err)
		}
	}
}

File: scratch/inspect_updated_bookings.py
import sqlite3
import os
import sys

paths = [
    "C:/Users/hofma/GolandProjects/HuhnLite-Wails/HuhnLite.db",
    "C:/Users/hofma/GolandProjects/HuhnLite-Wails/build/bin/HuhnLite.db",
    os.path.join(os.environ.get("APPDATA", ""), "HuhnLite-Wails", "HuhnLite.db")
]

for p in paths:
    print(f"\n=========================================")
    print(f"DATABASE: {p}")
    print(f"=========================================")
    if not os.path.exists(p):
        print("File not found.")
        continue
    try:
        conn = sqlite3.connect(p)
        cursor = conn.cursor()
        
        # Schema info
        cursor.execute("PRAGMA table_info(BUCHUNG)")
        cols = [row[1] for row in cursor.fetchall()]
        print(f"BUCHUNG columns: {cols}")
        
        cursor.execute("""
            SELECT ID, ID_HERDEN, BUCHUNGSDATUM, SILONR, FUTTERVERBRAUCHTIER, FUTTERKTAG, TIERBESTAND 
            FROM BUCHUNG 
            WHERE SILONR = 1 AND (FUTTERVERBRAUCHTIER > 0 OR FUTTERKTAG > 0)
            ORDER BY BUCHUNGSDATUM DESC LIMIT 10
        """)
        rows = cursor.fetchall()
        print(f"Found {len(rows)} matching updated rows:")
        for r in rows:
            print(f"  ID: {r[0]} | Herd: {r[1]} | Date: {r[2]} | Silo: {r[3]} | Verbrauch: {r[4]} | Ktag: {r[5]} | Bestand: {r[6]}")
            
        conn.close()
    except Exception as e:
        print(f"Error checking: {e}")
